### PR TITLE
upgrades and fixes

### PR DESCRIPTION
**Pull request by ghost**
- upgrade csproj
- fix json respons for about and health endpoints
- fix name for Build-info

### CODE REVIEW

File: src/Utilitywarehouse.OpsEndpoints/Owin/OpsEndpointsMiddleware.cs [hunk 1]
```diff
@@ -21,14 +21,16 @@ public class OpsEndpointsMiddleware
         private readonly RequestDelegate _next;
         private readonly JsonSerializer _serializer;
         private readonly Dictionary<string, HttpHandler> _handlers;
+        private readonly JsonSerializerSettings _serializerSettings;
 
         public OpsEndpointsMiddleware(RequestDelegate next, OpsEndpointsMiddlewareOptions options)
         {
             _next = next;
             _handlers = ConfigureHandlers(options);
-            var serializer = new JsonSerializer {ContractResolver = new CamelCasePropertyNamesContractResolver()};
```
Comment (thinktainer) [inline src/Utilitywarehouse.OpsEndpoints/Owin/OpsEndpointsMiddleware.cs:29]: does this cause the rendering to be upper case?

File: src/Utilitywarehouse.OpsEndpoints/Owin/OpsEndpointsMiddleware.cs [hunk 2]
```diff
@@ -21,14 +21,16 @@ public class OpsEndpointsMiddleware
...
-            serializer.Converters.Add(new StringEnumConverter());
-            _serializer = serializer;
+            _serializerSettings = new JsonSerializerSettings
+            {
+                Formatting = Formatting.Indented
```
Comment (thinktainer) [inline src/Utilitywarehouse.OpsEndpoints/Owin/OpsEndpointsMiddleware.cs]: is this really required?

Comment (thinktainer) [reply src/Utilitywarehouse.OpsEndpoints/Owin/OpsEndpointsMiddleware.cs]: Bear in mind that we want this to return quickly, and any overhead should be avoided.

File: src/Utilitywarehouse.OpsEndpoints/Owin/OpsEndpointsMiddleware.cs [hunk 3]
```diff
@@ -54,11 +56,8 @@ public OpsEndpointsMiddleware(RequestDelegate next, OpsEndpointsMiddlewareOption
                 AboutResponse response = op.HealthModel.About().ToAboutResponse();
                 context.Response.StatusCode = 200;
                 context.Response.ContentType = "application/json";
-                using (var writer = new StreamWriter(context.Response.Body, Encoding.UTF8))
```
Comment (thinktainer) [inline src/Utilitywarehouse.OpsEndpoints/Owin/OpsEndpointsMiddleware.cs:57]: is this what causes the BOM characters to go away?

File: src/Utilitywarehouse.OpsEndpoints/Owin/OpsEndpointsMiddleware.cs [hunk 4]
```diff
@@ -84,9 +80,12 @@ public OpsEndpointsMiddleware(RequestDelegate next, OpsEndpointsMiddlewareOption
 
         public Task Invoke(HttpContext context)
         {
-            if (context.Request.Path.StartsWithSegments("/__", out var _, out var segment))
+            PathString segment = null;
+
+            if (context.Request.Path.StartsWithSegments("/__", out segment))
```
Comment (thinktainer) [inline src/Utilitywarehouse.OpsEndpoints/Owin/OpsEndpointsMiddleware.cs]: why go for the extra verbosity?

File: test/Utilitywarehouse.OpsEndpoints.Tests/Utilitywarehouse.OpsEndpoints.Tests.csproj [hunk 5]
```diff
@@ -1,39 +1,44 @@
...
       <Version>1.0.0-alpha-20161104-2</Version>
       <PrivateAssets>All</PrivateAssets>
     </PackageReference>
     <PackageReference Include="Microsoft.NET.Test.Sdk">
-      <Version>15.0.0-preview-20161123-03</Version>
+      <Version>15.0.0-preview-20170217-05</Version>
     </PackageReference>
     <PackageReference Include="Newtonsoft.Json">
-      <Version>9.0.1</Version>
+      <Version>9.0.2-beta2</Version>
```
Comment (thinktainer) [inline test/Utilitywarehouse.OpsEndpoints.Tests/Utilitywarehouse.OpsEndpoints.Tests.csproj]: not keen on this update, is there anything this gives us, which 9.0.1 doesn't have?

File: uw-opsendpoints-dotnet.sln [hunk 6]
```diff
@@ -35,16 +35,16 @@ Global
...
 		{AB862E4C-C877-4EBB-83E9-81B60BC9AEE9}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
 		{AB862E4C-C877-4EBB-83E9-81B60BC9AEE9}.Debug|Any CPU.Build.0 = Debug|Any CPU
-		{AB862E4C-C877-4EBB-83E9-81B60BC9AEE9}.Debug|x64.ActiveCfg = Debug|x64
-		{AB862E4C-C877-4EBB-83E9-81B60BC9AEE9}.Debug|x64.Build.0 = Debug|x64
-		{AB862E4C-C877-4EBB-83E9-81B60BC9AEE9}.Debug|x86.ActiveCfg = Debug|x86
-		{AB862E4C-C877-4EBB-83E9-81B60BC9AEE9}.Debug|x86.Build.0 = Debug|x86
+		{AB862E4C-C877-4EBB-83E9-81B60BC9AEE9}.Debug|x64.ActiveCfg = Debug|Any CPU
+		{AB862E4C-C877-4EBB-83E9-81B60BC9AEE9}.Debug|x64.Build.0 = Debug|Any CPU
+		{AB862E4C-C877-4EBB-83E9-81B60BC9AEE9}.Debug|x86.ActiveCfg = Debug|Any CPU
+		{AB862E4C-C877-4EBB-83E9-81B60BC9AEE9}.Debug|x86.Build.0 = Debug|Any CPU
```
Comment (thinktainer) [inline uw-opsendpoints-dotnet.sln]: why do we flatten the build configurations to AnyCPU? I anticipate problems when used in [self contained deployments](https://docs.microsoft.com/en-us/dotnet/articles/core/preview3/deploying/).

**Review by thinktainer — CHANGES_REQUESTED** [review]
Other than comments LGTM

File: src/Utilitywarehouse.OpsEndpoints/Owin/OpsEndpointsMiddleware.cs (hunk 1 above)
Comment (ghost) [reply src/Utilitywarehouse.OpsEndpoints/Owin/OpsEndpointsMiddleware.cs:29]: I am using the JsonSerializerSettings so don't need the JsonSerializer object anymore.

File: src/Utilitywarehouse.OpsEndpoints/Owin/OpsEndpointsMiddleware.cs (hunk 2 above)
Comment (ghost) [reply src/Utilitywarehouse.OpsEndpoints/Owin/OpsEndpointsMiddleware.cs]: ok, I think we can remove the formatting

File: src/Utilitywarehouse.OpsEndpoints/Owin/OpsEndpointsMiddleware.cs (hunk 3 above)
Comment (ghost) [reply src/Utilitywarehouse.OpsEndpoints/Owin/OpsEndpointsMiddleware.cs:57]: yes, I think the StreamWriter is the one that adds the BOM characters.

File: src/Utilitywarehouse.OpsEndpoints/Owin/OpsEndpointsMiddleware.cs (hunk 4 above)
Comment (ghost) [reply src/Utilitywarehouse.OpsEndpoints/Owin/OpsEndpointsMiddleware.cs]: this changes we can undo. I did it before using VS2017

File: test/Utilitywarehouse.OpsEndpoints.Tests/Utilitywarehouse.OpsEndpoints.Tests.csproj (hunk 5 above)
Comment (ghost) [reply test/Utilitywarehouse.OpsEndpoints.Tests/Utilitywarehouse.OpsEndpoints.Tests.csproj]: yes, we can keep 9.0.1. It was probably updated automatically by nuget manager

File: uw-opsendpoints-dotnet.sln (hunk 6 above)
Comment (ghost) [reply uw-opsendpoints-dotnet.sln]: I had compatibility issues when adding the package to our projects. But this I think we can discuss it in more details....

File: src/Utilitywarehouse.OpsEndpoints/Owin/OpsEndpointsMiddleware.cs [hunk 7]
```diff
@@ -21,14 +21,12 @@ public class OpsEndpointsMiddleware
         private readonly RequestDelegate _next;
         private readonly JsonSerializer _serializer;
         private readonly Dictionary<string, HttpHandler> _handlers;
+        private readonly JsonSerializerSettings _serializerSettings;
```
Comment (thinktainer) [inline src/Utilitywarehouse.OpsEndpoints/Owin/OpsEndpointsMiddleware.cs]: It looks like these are not being used anywhere in the file, same as the JsonSerializer. Are these configured globally somewhere?

File: src/Utilitywarehouse.OpsEndpoints/Owin/OpsEndpointsMiddleware.cs (hunk 1 above)
Comment (thinktainer) [reply src/Utilitywarehouse.OpsEndpoints/Owin/OpsEndpointsMiddleware.cs:29]: It looks like these are not used anywhere in the file. can you confirm that the emitted json is now camelcased properly(lower case first letter)

Comment (ghost) [reply src/Utilitywarehouse.OpsEndpoints/Owin/OpsEndpointsMiddleware.cs:29]: Yes you are right, I added  CamelCase settings back.

File: src/Utilitywarehouse.OpsEndpoints/Owin/OpsEndpointsMiddleware.cs (hunk 7 above)
Comment (ghost) [reply src/Utilitywarehouse.OpsEndpoints/Owin/OpsEndpointsMiddleware.cs]: Using _serializerSettings again.

File: src/Utilitywarehouse.OpsEndpoints/project.json [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+      "imports": "dnxcore50"
+    }
+  },
+
+  "runtimes": {
+    "debian.8-x64": {},
+    "win81-x64": {},
+    "osx.10.12-x64": {}
+  }
+}
```
Comment (thinktainer) [inline src/Utilitywarehouse.OpsEndpoints/project.json]: Remove project.json please

**Review by thinktainer — APPROVED** [review]

**Pull request merged**